src/Uno.UI/UI/Xaml/DependencyObjectCollection.cs

```diff
@@ -205,7 +212,17 @@ internal List<T>.Enumerator GetEnumeratorFast()
 			=> _list.GetEnumerator();
 
 		private void RaiseVectorChanged(CollectionChange change, int index)
-			=> VectorChanged?.Invoke(this, new VectorChangedEventArgs(change, (uint)index));
+		{
+			if (_vectorChangedHandlers is not null)
+			{
+				var args = new VectorChangedEventArgs(change, (uint)index);
+
+				foreach (var handler in _vectorChangedHandlers)
+				{
+					handler.Invoke(this, args);
+				}
```

Comment:
This feels too dangerous tbh. I don't see any guarantee that the handler won't attempt to subscribe/unsubscribe from the event which would mutate the list.

Reply:
Agreed, there's a need for copying when executing if the list changes, though copying may cause a different set of trouble, will looking into it further.